File: backend/src/domain/entities/OrganizationMember.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  Index,
} from 'typeorm';

@Entity('organization_members')
@Index(['organizationId', 'userId'], { unique: true })
export class OrganizationMember {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'organization_id' })
  organizationId: number;

  @Column({ name: 'user_id' })
  userId: number;

  @Column({ type: 'varchar', name: 'display_name', nullable: true })
  displayName: string;

  @Column({ type: 'varchar', name: 'org_specific_avatar_url', nullable: true })
  orgSpecificAvatarUrl: string;

  @Column({ type: 'integer', name: 'verification_level', default: 0 })
  verificationLevel: number;

  @Column({ type: 'boolean', name: 'is_trusted_verifier', default: false })
  isTrustedVerifier: boolean;

  @Column({ type: 'varchar', default: 'Active' })
  status: string;

  @CreateDateColumn({ type: 'timestamp', name: 'joined_at' })
  joinedAt: Date;
}

File: backend/src/main/java/com/service/backend/domain/entity/OrganizationMember.java
package com.service.backend.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("organization_members")
public class OrganizationMember {

    @Id
    private Long id;

    @Column("organization_id")
    private Long organizationId;

    @Column("user_id")
    private Long userId;

    @Column("display_name")
    private String displayName;

    @Column("org_specific_avatar_url")
    private String orgSpecificAvatarUrl;

    @Column("verification_level")
    @Builder.Default
    private Integer verificationLevel = 0;

    @Column("is_trusted_verifier")
    @Builder.Default
    private Boolean isTrustedVerifier = false;

    @Builder.Default
    private String status = "Active";

    @CreatedDate
    @Column("joined_at")
    private LocalDateTime joinedAt;
}
